docs/plots/backends.rst

```diff
@@ -0,0 +1,362 @@
+************************
+Plotting backend details
+************************
+
+.. warning::
+   The plotting backend API is currently re-designed and details might change in the near future.
+
+Plotting in Sherpa is done through *plotting backends*. A *backend* is an
+external package that performs the actual plotting function such as :term:`matplotlib`, i.e. it creates a
+canvas and puts color on paper or on the screen. The Sherpa plotting classes are
+written in a backend-independent way; they work with any plotting backend that
+is supported by Sherpa. We also use the term *backend* to refer to Sherpa
+classes that connect the Sherpa plot objects with the plotting backend by
+translating Sherpa options into backend-specific commands. The details of that
+are explained below, but, in short, a backend class would translate the sherpa
+backend-independent "plot a line in red from here to there" into backend
+specific ``plt.plot(x,y, linecolor='r')`` (if the backend is :term:`matplotlib`).
+
+Instead of utilizing Sherpa classes and commands to plot data, users can
+alternatively just access the data in the Sherpa data objects (e.g. the x and y
+values of a dataset) and perform plotting operations directly with any plotting
+backend that is available to them. This method may be less convenient, but it
+works for ploting backends not (yet) supported by Sherpa.
+
+Which backend is used?
+======================
+
+When the `sherpa.plot` module is first imported, Sherpa tries to import the
+backends installed with Sherpa in the order listed in the
+``options.plot_pkg`` setting from the ``sherpa.rc`` startup file.
+The first module that imports successfully is set as the active
+backend. The following command prints the name and the location
+on disk of that module::
+
+   >>> from sherpa import plot
+   >>> print(plot.backend)
+
```

Comment:
So, with `pylab` selected and `matplotlib` installed I get

```
In [4]: print(plot.backend)
<sherpa.plot.pylab_backend.PylabBackend object at 0x7f09139257b0>
```

which doesn't quite match the text (or at least, it's not what I expected from "the name and the location on disk" of that module). I think this is likely a holdover from the old behavior.

If I've selected a plot package which is not installed (e.g. `pylab` when matplotlbi is not installed, or `bokeh` when we haven't yet merged that backend ;-) I get

```
In [2]: print(plot.backend)
<sherpa.plot.backends.BasicBackend object at 0x7f0840de1ab0>
```

Reply:
Good question. What do we want to display? All backends have a "name" attribute, that defaults to the class name ("BasicBackend" in this case) except for pylab, where the name is "pylab" because that's what it was before and I don't want to break the API. It's used, e.g. in ciao_contrib. I could implement `__str__` and return the name instead of the usual `<sherpa.plot.backends.BasicBackend object at 0x7f0840de1ab0>` instead.